registry/config.go

```diff
@@ -136,6 +136,63 @@ func NewServiceConfig(options *Options) *ServiceConfig {
 	return config
 }
 
+// ModifyRegistryService add or remove insecure registry
+func (config *ServiceConfig) ModifyRegistryService(method, value string) error {
+	if method == "Add" {
+		// Check if CIDR was passed to --insecure-registry
+		_, ipnet, err := net.ParseCIDR(value)
+		if err == nil {
+			// Valid CIDR.
+			for _, ipNet := range config.InsecureRegistryCIDRs {
+				if (*net.IPNet)(ipNet).String() == ipnet.String() {
+					return nil
+				}
+			}
+			config.InsecureRegistryCIDRs = append(config.InsecureRegistryCIDRs, (*netIPNet)(ipnet))
+		} else {
+			hostport := strings.Split(value, ":")
```

Comment:
Would like to see something a little more robust like `url.Parse` rather than making assumptions that could lead to bad configuration.